odiglet/pkg/ebpf/manager.go

```diff
@@ -0,0 +1,371 @@
+		return err
+	}
+
+	m.DetailsByPid[e.PID] = InstrumentationDetails{
+		Inst:              inst,
+		Pod:               types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name},
+		Lang:              lang,
+		WorkloadName:      workload.Name,
+		WorkloadNamespace: workload.Namespace,
+	}
+
+	m.Logger.Info("instrumentation loaded", "pid", e.PID, "pod", pod.Name, "container", containerName, "language", lang, "sdk", sdk)
+
+	// write instrumentation instance CR with success status
+	msg := fmt.Sprintf("Successfully loaded eBPF probes to pod: %s container: %s", pod.Name, containerName)
+	err = m.updateInstrumentationInstanceStatus(ctx, pod, containerName, workload, e.PID, InstrumentationHealthy, LoadedSuccessfully, msg)
+	if err != nil {
+		m.Logger.Error(err, "failed to update instrumentation instance for successful load")
+	}
+
+	go func() {
+		err := inst.Run(ctx)
+		if err != nil && !errors.Is(err, context.Canceled) {
+			m.Logger.Error(err, "failed to run instrumentation")
+			// these errors occur after the instrumentation is loaded
+			// write instrumentation instance CR with error status
+			err = m.updateInstrumentationInstanceStatus(ctx, pod, containerName, workload, e.PID, InstrumentationUnhealthy, FailedToRun, err.Error())
+			if err != nil {
+				m.Logger.Error(err, "failed to update instrumentation instance for failed instrumentation run")
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (m *Manager) instrumentationSDKConfig(ctx context.Context, w *workloadUtils.PodWorkload, lang common.ProgrammingLanguage, podKey types.NamespacedName) *odigosv1.SdkConfig {
+	instrumentationConfig := odigosv1.InstrumentationConfig{}
+	instrumentationConfigKey := client.ObjectKey{
+		Namespace: w.Namespace,
+		Name:      workloadUtils.CalculateWorkloadRuntimeObjectName(w.Name, w.Kind),
+	}
+	if err := m.Client.Get(ctx, instrumentationConfigKey, &instrumentationConfig); err != nil {
+		// this can be valid when the instrumentation config is deleted and current pods will go down soon
+		m.Logger.Error(err, "failed to get initial instrumentation config for instrumented pod", "pod", podKey.Name, "namespace", podKey.Namespace)
+		return nil
+	}
+	for _, config := range instrumentationConfig.Spec.SdkConfigs {
+		if config.Language == lang {
+			return &config
+		}
+	}
+	return nil
+}
+
+func (m *Manager) applyInstrumentationConfigurationForSDK(ctx context.Context, workloadKey types.NamespacedName, sdkConfig *odigosv1.SdkConfig) error {
+	var err error
+	for _, instDetails := range m.DetailsByPid {
+		if instDetails.WorkloadName != workloadKey.Name || instDetails.WorkloadNamespace != workloadKey.Namespace {
+			continue
+		}
+
+		if instDetails.Lang != sdkConfig.Language {
+			continue
+		}
+
+		m.Logger.Info("applying configuration to instrumentation", "workload", workloadKey, "pid", instDetails.Inst, "pod", instDetails.Pod)
+		err = errors.Join(err, instDetails.Inst.ApplyConfig(ctx, sdkConfig))
```

Comment:
I think the `ApplyConfig` call is the main and important part of this statement, and having it inside the `errors.Join` makes it harder to read.

How about?
```suggestion
                 applyErr := instDetails.Inst.ApplyConfig(ctx, sdkConfig)
		err = errors.Join(err, applyErr)
```